node/service/src/relay_chain_selection.rs

```diff
@@ -417,7 +417,7 @@ where
 
 		let lag = if cfg!(feature = "disputes") {
 			// Prevent sending flawed data to the dispute-coordinator.
-			if Some(subchain_block_descriptions.len() as _) !=
+			if Some(subchain_block_descriptions.len() as u32 - 1) !=
```

Comment:
Do we guarantee that block descriptions are always not-empty?
Would be nice to add a comment why -1.
```suggestion
			if subchain_block_descriptions.len().checked_sub(1) as _ !=
```